core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/function/Min.java

```diff
@@ -16,80 +16,54 @@
 
 package io.cdap.plugin.batch.aggregator.function;
 
+import io.cdap.cdap.api.data.format.StructuredRecord;
 import io.cdap.cdap.api.data.schema.Schema;
 
-import javax.annotation.Nullable;
-
 /**
  * Calculates minimum values of a field in a group.
  */
-public class Min extends NumberFunction {
-  private Integer minInt;
-  private Long minLong;
-  private Float minFloat;
-  private Double minDouble;
+public class Min extends NumberFunction<Min> {
 
-  public Min(String fieldName, @Nullable Schema fieldSchema) {
+  public Min(String fieldName, Schema fieldSchema) {
     super(fieldName, fieldSchema);
   }
 
   @Override
-  protected void startInt() {
-    minInt = null;
-  }
-
-  @Override
-  protected void startLong() {
-    minLong = null;
+  public void mergeValue(StructuredRecord record) {
+    combine(record.get(fieldName));
   }
 
   @Override
-  protected void startFloat() {
-    minFloat = null;
+  public void mergeAggregates(Min otherAgg) {
+    combine(otherAgg.getAggregate());
   }
 
-  @Override
-  protected void startDouble() {
-    minDouble = null;
-  }
-
-  @Override
-  protected void updateInt(int val) {
```

Comment:
I think these were originally done so that we don't do an extra comparison on type for every record. I don't think it actually matters much from a performance perspective, don't imagine this was a bottleneck in any way. Are we just changing it for readability?

Reply:
Yes, the original class is really hard to understand and it creates delegates to make sure the correct function is called. Since this is just a simple comparison, I want to remove it for better readability.